python/lsst/sims/catUtils/utils/LightCurveGenerator.py

```diff
@@ -0,0 +1,496 @@
+
+        Output:
+        -------
+        A dict of light curves.  The dict is keyed on the object's uniqueId.
+        This yields a dict keyed on bandpass, which yields a dict keyed on
+        'mjd', 'mag', and 'error', i.e.
+
+        output[111]['u']['mjd'] is a numpy array of the MJD of observations
+        of object 111 in the u band.
+
+        output[111]['u']['mag'] is a numpy array of the magnitudes of
+        object 111 in the u band.
+
+        output[111]['u']['error'] is a numpy array of the magnitude uncertainties
+        of object 111 in the u band.
+
+        And a dict of truth data for each of the objects (again, keyed on
+        uniqueId).  The contents of this dict will vary, depending on the
+        variability model being used, but should be sufficient to reconstruct
+        the actual light curves 'by hand' if necessary (or determine the true
+        period of a variable source, if attempting to evaluate the performance
+        of a classification scheme against a proposed observing cadence).
+        """
+
+        # First get the list of ObservationMetaData objects corresponding
+        # to the OpSim pointings in the region and bandpass of interest
+
+        t_start = time.time()
+
+        self.mjd_dict = {}
+        self.mag_dict = {}
+        self.sig_dict = {}
+        self.band_dict = {}
+        self.truth_dict = {}
+
+        cat_dict = {}
+        for grp in pointings:
+            for obs in grp:
+                if obs.bandpass not in cat_dict:
+                    cat_dict[obs.bandpass] = self._lightCurveCatalogClass(self._catalogdb, obs_metadata=obs)
+
+        # Loop over the list of groups ObservationMetaData objects,
+        # querying the database and generating light curves.
+        for grp in pointings:
+
+            self._mjd_min = grp[0].mjd.TAI
+            self._mjd_max = grp[-1].mjd.TAI
+
+            print('starting query')
+
+            t_before_query = time.time()
+            query_result = self._get_query_from_group(grp, chunk_size)
+
+            print('query took ', time.time()-t_before_query)
+
+            self._light_curves_from_query(cat_dict, query_result, grp)
+
+        output_dict = {}
+        for unique_id in self.mjd_dict:
+            output_dict[unique_id] = {}
+            for bp in self.mjd_dict[unique_id]:
+
+                output_dict[unique_id][bp] = {}
+
+                # we must sort the MJDs because, if an object appears in multiple
+                # spatial pointings, its observations will be concatenated out of order
+                mjd_arr = np.array(self.mjd_dict[unique_id][bp])
+                mjd_dexes = np.argsort(mjd_arr)
+
+                output_dict[unique_id][bp]['mjd'] = mjd_arr[mjd_dexes]
+                output_dict[unique_id][bp]['mag'] = np.array(self.mag_dict[unique_id][bp])[mjd_dexes]
+                output_dict[unique_id][bp]['error'] = np.array(self.sig_dict[unique_id][bp])[mjd_dexes]
+
```

Comment:
Yeah... or just to give users a heads up if things are going slower than they intended.  I will eventually take them out.